src/pybamm/solvers/summary_variable.py

```diff
@@ -123,7 +128,7 @@ def __getitem__(self, key: str) -> float | list[float]:
             # return it if it exists
             return self._variables[key]
         elif key == "Cycle number":
-            return self.cycle_number
+            return cast(list[float], self.cycle_number.tolist())
```

Comment:
Don't `cast` here. Remember, `tolist()` is an expensive operation as it converts a NumPy `array` (which is C underneath) to a Python `list`. You should instead update the function's signature to include `npt.NDArray` as a possible return type.